platform-sdk/swirlds-virtualmap/src/jmh/java/com/swirlds/virtualmap/benchmark/reconnect/ReconnectHalfMillionNodesBench.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.virtualmap.benchmark.reconnect;
+
+import com.swirlds.common.constructable.ConstructableRegistryException;
+import com.swirlds.virtualmap.test.fixtures.TestKey;
+import com.swirlds.virtualmap.test.fixtures.TestValue;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+@BenchmarkMode(Mode.AverageTime)
+@State(Scope.Thread)
+public class ReconnectHalfMillionNodesBench extends VirtualMapReconnectBenchBase {
+
+    private static final Map<TestKey, TestValue> testTeacherMap = new HashMap<>();
+    private static final Map<TestKey, TestValue> testLearnerMap = new HashMap<>();
+
+    static {
+        try {
+            VirtualMapReconnectBenchBase.startup();
+        } catch (ConstructableRegistryException e) {
+            throw new RuntimeException(e);
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Create a state to be reused in every run
+        StateBuilder.buildState(new Random(9823452658L), 500_000, 0.15, 0.15, testTeacherMap::put, testLearnerMap::put);
```

Comment:
All these hard-coded values should be changed to JMH params, so they can be changed without recompiling code

Reply:
The JMH parameter values would only be available at setupEach() which runs per every invocation of the benchmark. I moved the construction of the state into the static{} block by design in order to avoid wasting time reconstructing the same key/values for the map for every test invocation. So this particular benchmark would continue to use these hard-coded parameters, and potentially we'll even increase the state size here which will make it even more important for this particular benchmark.

At a later time, we can create a separate test/gradle target using the same base class where the values would be parametrized. You'd probably need to use smaller states, or reduce the number of iterations (and hence reduce the confidence of the measurements) in order to support that though. But we can figure this out when we introduce that separate benchmark implementation.